src/js/Sort/TableSorter.js

```diff
@@ -110,7 +75,7 @@ class TableSorter {
 		const intlCollator = getIntlCollator();
 		const cellFormatter = this._cellFormatter;
 		const type = tableHeaderElement.getAttribute('data-o-table-data-type') || undefined;
-		table.tableRows.sort((a, b) => {
+		table.tableRows = table.tableRows.sort((a, b) => {
```

Comment:
The assignment is not required because `Array.prototype.sort` is a mutative method, it modifies the array in-place, it doesn't return a new array.
```
> a = [1,4,3,5,7,8]
[ 1, 4, 3, 5, 7, 8 ]
> a.sort()
[ 1, 3, 4, 5, 7, 8 ]
> a
[ 1, 3, 4, 5, 7, 8 ]
> 
```